www/js/app.js

```diff
@@ -650,25 +751,11 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
         }
         $('input:radio[name=contentInjectionMode]').prop('checked', false);
         $('input:radio[name=contentInjectionMode]').filter('[value="' + value + '"]').prop('checked', true);
-        contentInjectionMode = value;
         // Save the value in the Settings Store, so that to be able to keep it after a reload/restart
-        settingsStore.setItem('lastContentInjectionMode', value, Infinity);
+        settingsStore.setItem('contentInjectionMode', value, Infinity);
         refreshCacheStatus();
+        refreshAPIStatus();
     }
-            
-    // At launch, we try to set the last content injection mode (stored in Settings Store)
-    var lastContentInjectionMode = settingsStore.getItem('lastContentInjectionMode');
-    if (lastContentInjectionMode) {
-        setContentInjectionMode(lastContentInjectionMode);
-    }
-    else {
-        setContentInjectionMode('jquery');
-    }
-    
-    var serviceWorkerRegistration = null;
-    
-    // We need to establish the caching capabilities before first page launch
-    refreshCacheStatus();
```

Comment:
We now instantiate the Service Worker towards the top of app.js, in the block "Set the State and UI Settings associated with parameters".